From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: deepak1556 <hop2deep@gmail.com>
Date: Tue, 29 Mar 2022 20:17:54 +0900
Subject: ui_gtk_public_header.patch

Allow electron to depend on gtk_util.h and gtk_ui.h from //ui/gtk/

diff --git a/ui/gtk/BUILD.gn b/ui/gtk/BUILD.gn
index db29cdca6f02fb8ca9dd0f25e13ed902589adc25..48f19e145cb81f4e4d2fdd6f3bd1cc72128358a7 100644
--- a/ui/gtk/BUILD.gn
+++ b/ui/gtk/BUILD.gn
@@ -39,7 +39,11 @@ generate_stubs("gtk_stubs") {
 }
 
 component("gtk") {
-  public = [ "gtk_ui_factory.h" ]
+  public = [
+    "gtk_ui.h",
+    "gtk_ui_factory.h",
+    "gtk_util.h",
+  ]
 
   sources = [
     "gtk_color_mixers.cc",
@@ -49,11 +53,9 @@ component("gtk") {
     "gtk_key_bindings_handler.cc",
     "gtk_key_bindings_handler.h",
     "gtk_ui.cc",
-    "gtk_ui.h",
     "gtk_ui_factory.cc",
     "gtk_ui_platform.h",
     "gtk_util.cc",
-    "gtk_util.h",
     "input_method_context_impl_gtk.cc",
     "input_method_context_impl_gtk.h",
     "native_theme_gtk.cc",
